### DISCUSSION TITLE
Auto-Save for any new Draft post

### DISCUSSION THREAD

**Discussion by cmgorton** [forem/forem#15862] [Icebox]
Originally posted on forem.dev by [Jayendran Arumugam](https://forem.dev/jayendran/feature-request-auto-save-for-any-new-draft-post-5b3g) 

During my post creation in dev.to, in the draft mode accidentally I forget to click the save button. The next day when I saw my yesterday's contribution was gone and the draft has only the version which I saved much earlier.

It would be great if you can provide an auto-save feature for any draft version on the post. So that it won't be a big shock like me for others in the future!

**Comment by cmgorton** [comment]
Our product team let me know they like this idea but it's going to be an expensive feature so we will want to get SRE involved in this discussion.

**Reply by thomasbnt** [reply]
Oooh totaly agreed to this idea!

**Comment by Xirdim** [comment]
Though I like this idea, I'm afraid that some people want to decide whether to save the contribution or not, by themselves each time. So I think it may be better if each user can choose whether to use the feature.

**Comment by szabgab** [comment]
I am not sure why would this be expensive?

When the user starts editing an article a JavaScript function could be scheduled to run every N seconds. It would copy the content of the editor to a some variable in the memory. Every time it runs it could compare the current content of the editor with the one in this variable. If something has changed only then would it send it to the server. It would then also update the copy in this variable.

**Reply by thomasbnt** [reply]
And save that in localStorage ?

**Reply by szabgab** [reply]
It could be stored in the memory of the browser. I don't think there is a need to store it in local storage.
